### PR TITLE
Updates to NvFaidx, Fasta Noodles, and Sequence Accessor

### PR DESCRIPTION
**Pull request by skothenhill-nv**
# Overview
Facilitates iterating over the NvFaidx object as though it were a dictionary. adds rust-level implementations of common DNA transformations. Shows runtime improvements over a common alternative, BioPython. Benchmarking is a test marked as skip to avoid including the Biopython dependency in release images. 

# Timings
Runtime estimates were computed by executing functions over a random 100 sequence 10000 nt set of sequences.

| Function         | Biopython Runtime | Bionemo-Noodles Runtime | Speedup (Biop / Bionemo-Noodles) |
|------------------|------------------:|-------------------------:|--------------------------------:|
| reverse          | 5.86e-4          | 1.98e-5                 | 29.6                            |
| transcribe       | 1.25e-3          | 4.20e-5                 | 29.7                            |
| back_transcribe  | 9.42e-5          | 8.82e-6                 | 10.7                            |
| complement       | 5.46e-4          | 8.42e-5                 | 6.49                            |

### CODE REVIEW

**Review by cspades — APPROVED** [review]
LGTM

File: docs/docs/user-guide/examples/bionemo-esm2/finetune.md [hunk 1]
```diff
@@ -230,22 +230,22 @@ We download a CSV example dataset of articical sequences for this inference exam
...
+DATA_PATH=$(download_bionemo_data esm2/testdata_esm2_infer:2.0)
+RESULTS_PATH=$WORKDIR/esm2_finetune_tutorial/
 
 infer_esm2 --checkpoint-path <finetune checkpoint path> \
            --data-path $DATA_PATH \
            --results-path $RESULTS_PATH \
            --config-class ESM2FineTuneSeqConfig
 ```
 
-This will create a result `.pt` file under `$WORKDIR/esm2_finetune_tutorial/inference_results.pt` which can be loaded via PyTorch library in python environment:
```
Comment (farhadrgh) [inline docs/docs/user-guide/examples/bionemo-esm2/finetune.md:242]: Thanks, I missed this

**Review by farhadrgh — APPROVED** [review]

**Review by jstjohn — APPROVED** [review]

File: sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py [hunk 2]
```diff
@@ -33,7 +33,15 @@ class SequenceAccessor:
     """
 
     def __init__(self, reader: PyIndexedMmapFastaReader, seqid: str, length: int) -> None:
-        """Construct a SequenceAccessor object.
+        """Construct a SequenceAccessor object. Ultimately this is used as a convenience object with NvFaidx.
+
+        When querying the following are true:
+            - Negative indexing is supported, but it does not wrap. so query[-10000] for a sequence of length 1 will fail.
+            - out of bounds indexing is truncated: query[1:999999999] will return a string from position 1 to the terminus.
```
Comment (edawson) [inline sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py:40]: What is the behavior if the start of the interval is also beyond the end of the string? Is empty string returned or is this a failure?

**Review by edawson — APPROVED** [review]
Approved. One comment asking for clarification of edge case behavior.

File: sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py (hunk 2 above)
Comment (skothenhill-nv) [reply sub-packages/bionemo-noodles/src/bionemo/noodles/nvfaidx.py:40]: added this to the docstring and a supporting test. (it returns the empty string).

**Pull request merged**